package.json

```diff
@@ -209,10 +208,10 @@
     "webpack-http-server": "^0.5.0"
   },
   "peerDependencies": {
-    "typescript": ">= 4.8.x"
+    "graphql": ">=16.8.1"
```

Comment:
Sadly, this isn't going to work. There isn't really a thing as optional peer dependency in JavaScript right now. You can think you defined one in `peerDependencies`, but that's just the installation part.

There's no way to _consume_ an optional peer dependency in the library's code. Dynamic imports aren't supported everywhere, and requires cause issues with ESM. 

We've tried this in the past to no avail, always meeting some edge case from our users. 

Sharing some context in case you will be more successful than me:

- #2187
- #2249
- #2267